app/lib/provider_discovery.rb

```diff
@@ -22,6 +22,7 @@ def discover_provider(url, options = {})
         format ||= :xml if provider_endpoint
       end
 
+      raise OEmbed::NotFound, url unless provider_endpoint
```

Comment:
Is there a time when `provider_endpoint` returns `''`? Considering that, `if provider_endpoint.present?` might be better.